### PR TITLE
chore: add license header to files and github action workflow to check license

### LINKED ISSUE #322: Check the license header for notation-go and its dependencies
### Is your feature request related to a problem?

Consider if notation-go needs dependency license checking and reporting

### What solution do you propose?

Consider adding dependency license checking and reporting similar to https://github.com/notaryproject/notation/issues/706

### What alternatives have you considered?

If notation-go does not require/need this, clearly document in README or other public location

### Any additional context?

_No response_

### PR DESCRIPTION
**Pull request by Two-Hearts**
Resolves #322 

This PR should be reviewed/merged after https://github.com/notaryproject/notation-core-go/pull/154.

### CODE REVIEW

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: .github/workflows/license-checker.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: License Checker
+
+on:
+  push:
+    branches:
+      - main
+      - release-*
```
Comment (shizhMSFT) [inline .github/workflows/license-checker.yml:20]: We don't have release branch for libraries.

File: .github/workflows/license-checker.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  pull_request:
+    branches:
+      - main
+      - release-*
```
Comment (shizhMSFT) [inline .github/workflows/license-checker.yml:24]: We don't have release branch for libraries.

**Review by shizhMSFT — APPROVED** [review]
LGTM. This PR must be merged after https://github.com/notaryproject/notation-core-go/pull/154

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Pull request merged**